ouroboros-network-framework/src/Ouroboros/Network/Context.hs

```diff
@@ -22,7 +22,8 @@ import Ouroboros.Network.PeerSelection.LedgerPeers.Type
 data ExpandedInitiatorContext addr m = ExpandedInitiatorContext {
     eicConnectionId    :: !(ConnectionId addr),
     eicControlMessage  :: !(ControlMessageSTM m),
-    eicIsBigLedgerPeer :: !IsBigLedgerPeer
+    eicIsBigLedgerPeer :: !IsBigLedgerPeer,
+    eicIsLocalRootPeer :: !Bool
```

Comment:
If I recall we are interested in marking local trusted peers, not just local ones. Am I right?  If that's true the field should be `eicIsLocalTrustedRootPeer`.

We should also provide a data type, instead a boolean:
```
data IsTrustedLocalRootPeer
 ...
```